shacl12-vocabularies/shnex-sparql.ttl

```diff
@@ -0,0 +1,901 @@
+## W3C SOFTWARE AND DOCUMENT NOTICE AND LICENSE
+## https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document
+## --------
+
+# W3C SHACL Node Expressions Vocabulary
+
+# THIS VERSION IS UNDER DEVELOPMENT BY THE DATA-SHAPES (SHACL 1.2) WG
+
+@prefix owl:   <http://www.w3.org/2002/07/owl#> .
+@prefix rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd:   <http://www.w3.org/2001/XMLSchema#> .
+@prefix dc:      <http://purl.org/dc/terms/> .
+@prefix sh:    <http://www.w3.org/ns/shacl#> .
+@prefix shnex: <http://www.w3.org/ns/shnex#> .
+@prefix shnex-sparql: <http://www.w3.org/ns/shnex-sparql#> .
+@prefix sparql:  <http://www.w3.org/ns/sparql#> .
+
+shnex-sparql:
+	a owl:Ontology ;
+	rdfs:label "W3C SHACL SPARQL Functions Vocabulary"@en ;
+	rdfs:comment "This vocabulary defines SPARQL functions for use with SHACL."@en ;
+	sh:declare [
+		sh:prefix "shnex" ;
+		sh:namespace "http://www.w3.org/ns/shnex#"^^xsd:anyURI ;
+	] ;
+	owl:imports <http://www.w3.org/ns/shnex#>, <http://www.w3.org/ns/sparql-query#>  .
+
+
+# Spec link pattern:
+# For functions in the source whose rdfs:isDefinedBy is the generic spec homepage
+# <http://www.w3.org/TR/sparql12-query/> a per-function anchor was constructed
+# as <http://www.w3.org/TR/sparql12-query/#func-LOCALNAME> where LOCALNAME is
+# the SPARQL local name (e.g., sparql:str -> #func-str). Where a function had a
+# more specific anchor in the source, the same target was used. (This comment
+# documents the fragment pattern used for constructed links.)
+
+# Imported SPARQL function terms (rdf:type, rdfs:label, dc:description, shnex:category, rdfs:isDefinedBy)
+# Category values: algebraic, numeric, string, date, aggregate, other
+
+sparql:plus
+	a sparql:Function ;
```

Comment:
Pending our final design, I believe this would become sh:ListParameterExpressionFunction.

Reply:
this I took directly from https://github.com/w3c/sparql-query/blob/main/sparql-ns.ttl. Should I make them all `a sh:ListParameterExpressionFunction` instead?

Reply:
I believe yes, but we may change the final design later. We could also leave it like this and make sparql:Function a subclass of sh:ListParameterExpressionFunction. So for now, maybe just leave it.